### ISSUE TITLE
Reafctor unit system?

### ISSUE THREAD

**Issue by joakim-hove** [OPM/opm-common#950]
We have started a long haul to add all the missing Eclipse100 keywords, and in that process things pop up - of course. Right now is the `WAPI`keyword where specific gravity of the injection stream can be set with [API Units](http://www.petroleum.co.uk/api) - that relationship totally breaks the current unit system.

The unit system has always felt a bit klunky, so this *could* be seen as an opportunity to refactor it? For now the density item in the `WAPI` keyword is just entered without dimension: https://github.com/OPM/opm-common/pull/945

**Comment by atgeirr** [comment]
I think refactoring to take into account the *utterly weird* API units should only be done if considered mission critical. Unfortunately, it looks to me like the only way to input data for using API tracking (which is essentially "choose PVT table depending on the average surface density of the oil present in the cell") requires this. On the other side, it seems to be somewhat isolated, so treating it as a separate dimension (i.e. not converting to SI densities) may be possible?

**Comment by joakim-hove** [comment]
> On the other side, it seems to be somewhat isolated, so treating it as a separate dimension (i.e. not converting to SI densities) may be possible?

Yes - or doing unit conversion manually in the `EclipseState` layer. Will not do anything right away at least.
